pkg/controller/applicationlayer/applicationlayer_controller.go

```diff
@@ -271,22 +267,29 @@ func (r *ReconcileApplicationLayer) Reconcile(ctx context.Context, request recon
 		}
 	}
 
-	lcSpec := applicationLayer.Spec.LogCollection
+	// prior client.Patch may nil out missing fields (in cases of outdated crd spec).
+	// We need to re-fill out defaults again. We don't need to do this if CRD is updated to match.
+	updateApplicationLayerWithDefaults(alp)
```

Comment:
I don't think this is needed. We can assume if there are updated CRDs that go along with an updated operator, they will be delivered together.